operator/src/lib.rs

```diff
@@ -91,6 +95,50 @@ fn count_pods(hashed: &HashMap<String, Vec<Pod>>) -> usize {
 }
 
 impl SparkState {
+    /// Process the cluster status:
+    /// If no status is set, write image, version and timestamp to status
+    /// Check for restart, start, stop commands
+    pub async fn process_cluster_status(&mut self) -> SparkReconcileResult {
+        if let Some(status) = &self.status {
+            if status.version != self.spec.version {
+                info!(
+                    "Current version [{}] - required version [{}] -> updating cluster...",
+                    status.version, &self.spec.version
+                );
+
+                for pod in &self.context.list_pods().await? {
+                    info!("Deleting pod [{}]", Meta::name(pod));
```

Comment:
Could you add a comment to the log saying _why_ the pod is being deleted with information about the versions involved